XenAdmin/TabPages/ManageUpdatesPage.cs

```diff
@@ -1307,5 +1312,23 @@ private void byHostToolStripMenuItem_Click(object sender, EventArgs e)
             dataGridViewHosts.Visible = true;
             Rebuild();
         }
+
+        private void toolStripButtonUpdate_Click(object sender, EventArgs e)
+        {
+            Program.Invoke(Program.MainWindow, () =>

```

Comment:
Why do we need Invoke here? Is it likely this won't be on the UI thread? (I also see the same in line 919?)